### ISSUE TITLE
Multi-processing or memory problems during the evaluation of classification

### ISSUE THREAD

**Issue by hustzeyu** [JDAI-CV/fast-reid#503]
Hi~ Thank you for the great work.

I have tried your FastClas project recently, while I have changed the ant_bee dataset into my large scale dataset which contains over 50,000 ids and more than 150,000 images for val along. It was so weird that I did not meet any problems at the training part(50,000 ids and over 2,000,000 images). However, I am not sure whether it is because you only use evaluate.reset() after each test_loader, or other reasons. I encounter the following problem:

 
![image](https://user-images.githubusercontent.com/16651561/120739375-fd5a1a80-c523-11eb-9fb7-5779a71b321f.png)

I have add my shm size to 64G, which I guess is more than enough. And I have also tried to ignore the warning by adding -W ignore, but the problem still occurs...

Do you have any ideas about this problem?

Thank you for reviewing.

**Comment by L1aoXingyu** [comment]
The problem may come from here
https://github.com/JDAI-CV/fast-reid/blob/8f8cbf94116b365f34e4971ac68f0fb01d33aab0/fastreid/evaluation/clas_evaluator.py#L58

It will save all the prediction logits, the gather them all on the main process. In your case, it will save 2000000 x 50000 matrix. It's so big.
I will fix it later and just compute the accuracy in each batch and then gather the accuracy to get the total accuracy.

**Comment by L1aoXingyu** [comment]
Try it again.

**Comment by hustzeyu** [comment]
> The problem may come from here
> https://github.com/JDAI-CV/fast-reid/blob/8f8cbf94116b365f34e4971ac68f0fb01d33aab0/fastreid/evaluation/clas_evaluator.py#L58
> 
> It will save all the prediction logits, the gather them all on the main process. In your case, it will save 2000000 x 50000 matrix. It's so big.
> I will fix it later and just compute the accuracy in each batch and then gather the accuracy to get the total accuracy.

Thank you 宇哥～
